becquerel/core/fitting.py

```diff
@@ -624,6 +647,116 @@ def fit(self, backend="lmfit"):
                 calc_covar=False,
             )  # no, bounds, default would be L-BFGS-B'
             # TODO: Calculate errors breaks minimization right now
+
+        elif self.backend in ["iminuit", "minuit"]:
+            # Translate a model from lmfit to minuit
+
+            # Poisson loss given the model specified by args
+            def model_loss(*args):
+                # Convert args to kwargs as lmfit.model.eval _requires_ kwargs,
+                # while (I think) the cost function passed to the Minuit object
+                # _requires_ args.
+                # TODO: check if the above is true for Minuit
+                # TODO: this feels dangerous!
+                # TODO: test out-of-order kwargs
+                kwargs = {self.model.param_names[i]: arg for i, arg in enumerate(args)}
+                y_eval = self.model.eval(x=self.x_roi, **kwargs)
+                if self.dx_roi is not None:
+                    y_eval *= self.dx_roi
+                lp = poisson_loss(y_eval, self.y_roi)
```

Comment:
We should warn if the data are not counts (e.g., subtracted spectrum).

Reply:
@markbandstra since we don't have access to the full `Spectrum` at this point, I wrote a little test:
```py
def _is_count_like(y):
    """Determine whether data 'looks like' it could be counts.

    Parameters
    ----------
    y : array-like
        Y-data
    """
    if np.any(y) < 0:
        return False
    if not np.allclose(y.astype(np.int), y):
        return False
    return True
```

Then I raise a `FittingWarning` if non-count-like data is passed into the Poisson loss fitter.

Reply:
Should it be `np.any(y < 0)`?

Reply:
Yes, thanks

Reply:
Seems like a sensible way to test for count data, setting aside a larger solution that involves tracking a "count" state within `Spectrum`.

Reply:
```
(y + 0.5).astype(int)
```

I don't think we need np.int specifically. Also this prevents weird rounding errors if the float is just a bit below the actual int.

Reply:
It feels like we had this discussion before, but I couldn't find it in a quick search. The fitting module uses:

```
np.rint(in_spectra).astype(int)
```

Reply:
Ok, I like the `np.rint` solution.